cypress/integration/unit_tests/polygon_draw_test.js

```diff
@@ -205,208 +263,466 @@ describe('Unit test for ShapeData', () => {
+      }
+    });
+    // We can click on the old polygon and view its notes,
+    cy.get('#test-map-div').click();
+    cy.get('#test-map-div').contains(notes).should('be.visible');
+    // And the new polygon and view its notes.
+    cy.get('#test-map-div').click(200, 300);
+    cy.get('#test-map-div').contains(newNotes).should('be.visible');
+
+    // Now hide both polygons, and verify that they're really gone.
+    setUserFeatureVisibilityInCypressAndAssert(false, true);
+    cy.get('#test-map-div').contains(notes).should('not.be.visible');
+    // Notes is invisible even if we click on the polygon, so it's really gone.
+    cy.get('#test-map-div').click(200, 300);
+    cy.get('#test-map-div').contains(newNotes).should('not.be.visible');
+  });
+
+  /**
+   * Draws a polygon with the given path, waits for it to save to Firestore,
+   * asserts that the write succeeded, clicks on the polygon, and asserts that
+   * the popup has the desired default data.
+   * @param {Array<LatLngLiteral>} polygonPath
+   * @return {Cypress.Chainable<void>}
+   */
+  function drawPolygonAndClickOnIt(polygonPath = path) {
+    drawPolygon(polygonPath);
+    return getCurrentUpdatePromiseInCypress().then(
+        () => assertOnFirestoreAndPopup(polygonPath, defaultData));
   }
 
-  /** Does nothing. */
-  setPendingCalculation() {}
+  /**
+   * Triggers the creation of a polygon with the given path.
+   * @param {Array<LatLngLiteral>} polygonPath Defaults to {@link path}
+   * @return {Cypress.Chainable<void>}
+   */
+  function drawPolygon(polygonPath = path) {
+    return cy.wrap(null).then(() => {
+      event.overlay = new google.maps.Polygon({
+        map: map,
+        paths: polygonPath,
+      });
+      google.maps.event.trigger(drawingManager, 'overlaycomplete', event);
+    });
+  }
 
   /**
-   * Sets calculatedData property.
-   * @param {Object} calculatedData
+   * Asserts that all Firestore writes have completed, that Firestore data
+   * matches expected data, and calls {@link assertOnPopup}.
+   * @param {Array<LatLngLiteral>} path
+   * @param {ExpectedData} expectedData
+   * @return {Cypress.Chainable}
    */
-  setCalculatedData(calculatedData) {
-    this.calculatedData = calculatedData;
+  function assertOnFirestoreAndPopup(path, expectedData) {
+    cy.wrap(null)
+        .then(() => expect(StoredShapeData.pendingWriteCount).to.eql(0))
+        .then(() => userShapes.get())
+        .then((querySnapshot) => {
+          expect(querySnapshot).to.have.property('size', 1);
+          const polygonDoc = querySnapshot.docs[0];
+          const firestoreId = polygonDoc.id;
+          expect(transformGeoPointArrayToLatLng(polygonDoc.get('geometry')))
+              .to.eql(path);
+          const calculatedData = Object.assign({}, expectedData);
+          delete calculatedData.notes;
+          expect(polygonDoc.get('calculatedData')).to.eql(calculatedData);
+          expect(polygonDoc.get('notes')).to.eql(expectedData.notes);
```

Comment:
confused about this copy to calculated data, removal of notes, and then assertion on notes that were copied and deleted